commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/util/ProductHelperImpl.java

```diff
@@ -77,8 +80,20 @@ public PriceModel getPrice(
 
 		if (discountValue != null) {
 			CommerceMoney discountAmount = discountValue.getDiscountAmount();
+			BigDecimal discountPercentage =
+				discountValue.getDiscountPercentage();
+			BigDecimal[] discountPercentages = discountValue.getPercentages();
+			CommerceMoney finalPrice = commerceProductPrice.getFinalPrice();
 
 			priceModel.setDiscount(discountAmount.format(locale));
+
+			priceModel.setDiscountPercentage(
+				_commercePriceFormatter.format(discountPercentage, locale));
+
+			priceModel.setDiscountPercentages(
+				_getFormattedDiscountPercentages(discountPercentages, locale));
+
+			priceModel.setFinalPrice(finalPrice.format(locale));
```

Comment:
no need to declare new variables here. `discountAmount` is only declared because we need to run `.format` on it after we get it. Oh, yeah `finalPrice` is the same. But `discountValue.getDiscountPercentage()` and `discountValue.getPercentages()` can just be passed in to the setter as-is